### PR TITLE
Add MFA documentation

### PR DESCRIPTION
**Pull request by hihuz**
Addresses: https://github.com/kontist/figure-backend/issues/5839

Please do let me know if the steps descriptions are not clear enough 🙏 

Have tested the curl examples against the prod API and was able to get a confirmed token.

I also made one small css change, as I found the headings were not visible enough in general except for the h2.

Here is how it looked before my change:
Screenshot 2019-12-03 at 10 38 37

And here is after:
Screenshot 2019-12-03 at 10 39 53

What do you think?

### CODE REVIEW

File: docs.markdown [hunk 1]
```diff
@@ -116,6 +116,140 @@ Response is again a JSON object, similar to the original one:
...
+  -X PATCH \
+  -d '{
+        "status": "VERIFIED"
+      }'
+```
+
+**Note:** in order for a challenge to be verified, this endpoint needs to be accessed using a *confirmed* token that you can obtain with [Device Binding](/docs/advanced-authentication#device-binding)
+
+> The above command returns `204 No Content` in case of success.
+
```
Comment (johannespfeiffer) [inline docs.markdown]: Setting the status to "VERIFIED" looks like a very special use case (only used by our mobile application and will be hard to do for others as only the mobile application will get the push notifications). Or do I miss sth. here?
I suggest to move that into the "advanced-authentication" page.

**Review by johannespfeiffer — APPROVED** [review]
Looks good, would move one section to advanced topics. Let's discuss that if you want to.

File: docs.markdown (hunk 1 above)
Comment (hihuz) [reply docs.markdown]: Yeah it is a very specific case and most likely not usable in any practical way I agree. I will definitely move it 👍

The case that I had in mind is that a dev could have an application where he implemented device binding (so the users can have confirmed access tokens there), and another one where he doesn't have device binding (e.g. a web app). 
In this case he could replicate the MFA flow that we have for our own web app, for instance by sending sms or his own push notification before / after hitting the endpoint to create the challenge, without the need to use the kontist app at all. Quite the edge case though yeah 😄

Comment (hihuz) [reply docs.markdown]: Have extracted that part here: 5457347b3131affea3da5e5283a215e2b51814bb

How does it look?

I'm unsure if we should reference it with a link on the main doc or not 🤔

Comment (johannespfeiffer) [reply docs.markdown]: Looks good and the use case you described makes sense. Would not add a special link in the main doc.

Comment (hihuz) [reply docs.markdown]: Yeah sounds good, I removed the link: 8b5f113e14241afeee6a04083a46a1309ac7ea31

Then I think we are good to merge it 🙂

**Pull request merged**